flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/CollectionFunctionsITCase.java

```diff
@@ -479,4 +480,82 @@ private Stream<TestSetSpec> arrayUnionTestCases() {
                                 "Invalid input arguments. Expected signatures are:\n"
                                         + "ARRAY_UNION(<COMMON>, <COMMON>)"));
     }
+
+    private Stream<TestSetSpec> arrayConcatTestCases() {
+        return Stream.of(
+                TestSetSpec.forFunction(BuiltInFunctionDefinitions.ARRAY_CONCAT)
+                        .onFieldsWithData(
+                                new Integer[] {1, 2, null},
+                                null,
+                                new Row[] {
+                                    Row.of(true, LocalDate.of(2022, 4, 20)),
+                                    Row.of(true, LocalDate.of(1990, 10, 14)),
+                                    null
+                                },
+                                1,
+                                new Integer[][] {{1}},
+                                new String[] {"123"})
+                        .andDataTypes(
+                                DataTypes.ARRAY(DataTypes.INT()),
+                                DataTypes.ARRAY(DataTypes.INT()),
+                                DataTypes.ARRAY(
+                                        DataTypes.ROW(DataTypes.BOOLEAN(), DataTypes.DATE())),
+                                DataTypes.INT(),
```

Comment:
I didn't get why this type `DataTypes.ARRAY(DataTypes.INT())` if it does not contain nulls?
it should be something like `DataTypes.ARRAY(DataTypes.INT().notNull()).notNull()`
similar notes about other types